src/platform/src/main/java/org/geoserver/util/IOUtils.java

```diff
@@ -373,24 +373,27 @@ public static File getZipOutputFile(File destDir, ZipEntry entry) throws IOExcep
     }
 
     public static void decompress(InputStream input, File destDir) throws IOException {
```

Comment:
Generally the PR looks good, but this change needs to be reverted, the responsibility of closing the stream provided as a parameter is on the caller of this method. Also, if the inputStream has been provided by a servelet, it should not be closed at all, it's the duty of the web container to do so

Reply:
Reverted